src/ansi-c/type2name.h

```diff
@@ -20,6 +20,15 @@ class namespacet;
 
 std::string type2name(const typet &type, const namespacet &ns);
 
+/**
+ * Constructs a string describing the given type, which can be used as part of
+ * a `C` identifier. The resulting identifier is not guaranteed to uniquely
+ * identify the type in all cases.
```

Comment:
This might need clarifying  - in fact, it might actually be worth making this explicit in the function name, e.g. rename it to something like `type_to_approximate_identifier` or `type_to_nonunique_identifier`.

Reply:
I get that "identifier is a bit of a misnomer in this case. However `nonunique_identifier` seems like a bit of an oxymoron to me. How does the term `partial_identifier` sound to you?

Reply:
`partial_identifer` sounds reasonable to me - it makes it clearer to an unsuspecting caller that they need to understand what is returning :-)

Reply:
Done.